zenoh/src/net/routing/dispatcher/face.rs

```diff
@@ -0,0 +1,245 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use super::super::router::*;
+use super::tables::TablesLock;
+use super::{resource::*, tables};
+use crate::net::primitives::Primitives;
+use std::any::Any;
+use std::collections::HashMap;
+use std::fmt;
+use std::sync::Arc;
+use zenoh_protocol::zenoh::RequestBody;
+use zenoh_protocol::{
+    core::{ExprId, WhatAmI, ZenohId},
+    network::{Mapping, Push, Request, RequestId, Response, ResponseFinal},
+};
+use zenoh_transport::multicast::TransportMulticast;
+#[cfg(feature = "stats")]
+use zenoh_transport::stats::TransportStats;
+
+pub struct FaceState {
+    pub(crate) id: usize,
+    pub(crate) zid: ZenohId,
+    pub(crate) whatami: WhatAmI,
+    pub(super) local: bool,
+    #[cfg(feature = "stats")]
+    pub(crate) stats: Option<Arc<TransportStats>>,
+    pub(crate) primitives: Arc<dyn crate::net::primitives::EPrimitives + Send + Sync>,
+    pub(crate) local_mappings: HashMap<ExprId, Arc<Resource>>,
+    pub(crate) remote_mappings: HashMap<ExprId, Arc<Resource>>,
+    pub(crate) next_qid: RequestId,
+    pub(crate) pending_queries: HashMap<RequestId, Arc<Query>>,
+    pub(crate) mcast_group: Option<TransportMulticast>,
+    pub(crate) hat: Box<dyn Any + Send + Sync>,
+}
+
+impl FaceState {
+    #[allow(clippy::too_many_arguments)]
```

Comment:
As general comment I believe it's better to address *clippy's lints* instead of allowing them.
If an exception is added, a proper justification should be provided in the code (e.g. in a comment).